library/std/src/sys/pal/unix/stack_overflow.rs

```diff
@@ -319,21 +319,29 @@ mod imp {
     ))]
     unsafe fn get_stack_start() -> Option<*mut libc::c_void> {
         let mut ret = None;
-        let mut attr: libc::pthread_attr_t = crate::mem::zeroed();
-        #[cfg(target_os = "freebsd")]
-        assert_eq!(libc::pthread_attr_init(&mut attr), 0);
+        let mut attr: mem::MaybeUninit<libc::pthread_attr_t> = if cfg!(target_os = "freebsd") {
+            let mut attr = mem::MaybeUninit::uninit();
+            assert_eq!(libc::pthread_attr_init((&raw mut attr) as *mut _), 0);
+            attr
```

Comment:
This moves `attr` after it has been initialized, which is not guaranteed to be sound.